pkg/jobs/hive/hive.go

```diff
@@ -335,11 +336,32 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
 		return getErr
 	}
 
-	if cvAvailableUpdates, ok := clusterVersion["status"].(map[string]interface{})["availableUpdates"].([]interface{}); ok {
-		for _, version := range cvAvailableUpdates {
-			if version.(map[string]interface{})["version"] == desiredUpdate {
-				clusterVersion["spec"].(map[string]interface{})["desiredUpdate"] = version
-				break
+	curatorAnnotations := curator.GetAnnotations()
+
+	if curatorAnnotations != nil && curatorAnnotations[ForceUpgradeAnnotation] == "true" {
```

Comment:
```suggestion
if curatorAnnotations != nil {
    if isForceUpgrade, ok := curatorAnnotations[ForceUpgradeAnnotation]; ok && isForceUpgrade == "true" {
```

You will need to check the existence of the key or it will panic crash I think.

Reply:
The original code is fine. If the first condition - curatorAnnotations is nil,  the second condition won't be checked in the if (condition1 && condition2)

Reply:
@xiangjingli you are right. Thanks.